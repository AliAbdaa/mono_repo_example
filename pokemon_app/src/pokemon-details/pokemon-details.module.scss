.details {
  display: flex;
  border-radius: 16px;
  width: 100%;
  padding: 50px;
  color: #009bcf;
  align-items: center;
  justify-content: center;
  font-size: 30px;
  font-weight: 300;
}

.imageWrapper {
  flex: 0 0 50%;
  text-align: center;
}

.image {
  transform: scale(3);
}

.pokemonInfo {
  flex: 0 0 50%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.types {

}

.id {
  font-weight: 700;
  font-size: 60px;
  text-shadow: 0 0 5px rgba(0, 192, 255, 0.97 );
}

.name {
  text-transform: uppercase;
  font-size: 40px;
}