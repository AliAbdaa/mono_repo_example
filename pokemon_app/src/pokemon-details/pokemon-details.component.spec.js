import React from 'react';
import { shallow } from 'enzyme';
import { PokemonDetails } from './pokemon-details.component';

describe('pokemon-details', () => {
  it('renders correctly', () => {
    const component = shallow(
      <PokemonDetails
        name={'test-name'}
        id={123}
        types={[
          {
            type: {
              name: 'test-type',
            },
          },
        ]}
        sprites={{
          front_default: 'test.jpg',
        }}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
