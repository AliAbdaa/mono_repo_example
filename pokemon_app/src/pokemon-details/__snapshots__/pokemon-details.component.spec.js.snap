// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemon-details renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonDetails
    id={123}
    name="test-name"
    sprites={
      Object {
        "front_default": "test.jpg",
      }
    }
    types={
      Array [
        Object {
          "type": Object {
            "name": "test-type",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <img
            alt="pokemon"
            src="test.jpg"
          />
        </div>,
        <div>
          <div>
            test-type
          </div>
          <div>
            123
          </div>
          <div>
            test-name
          </div>
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="pokemon"
            src="test.jpg"
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "pokemon",
            "className": undefined,
            "src": "test.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              test-type
            </div>,
            <div>
              123
            </div>,
            <div>
              test-name
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test-type",
              "className": undefined,
            },
            "ref": null,
            "rendered": "test-type",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 123,
              "className": undefined,
            },
            "ref": null,
            "rendered": 123,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test-name",
              "className": undefined,
            },
            "ref": null,
            "rendered": "test-name",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <img
              alt="pokemon"
              src="test.jpg"
            />
          </div>,
          <div>
            <div>
              test-type
            </div>
            <div>
              123
            </div>
            <div>
              test-name
            </div>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="pokemon"
              src="test.jpg"
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "pokemon",
              "className": undefined,
              "src": "test.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                test-type
              </div>,
              <div>
                123
              </div>,
              <div>
                test-name
              </div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test-type",
                "className": undefined,
              },
              "ref": null,
              "rendered": "test-type",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 123,
                "className": undefined,
              },
              "ref": null,
              "rendered": 123,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test-name",
                "className": undefined,
              },
              "ref": null,
              "rendered": "test-name",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
